### DISCUSSION TITLE
npm run build error   Jest worker encountered 1 child process exceptions, exceeding retry limit

### DISCUSSION THREAD

**Discussion by arafat813** [vercel/next.js#66234] [Help]
### Summary

PS E:\arafat2.0\blog> npm run build

> fati_blog@0.1.0 build
> next build

   ▲ Next.js 14.1.0
   - Environments: .env

   Creating an optimized production build ...
Compiler server unexpectedly exited with code: 3221225725 and signal: null

> Build error occurred
Error: Jest worker encountered 1 child process exceptions, exceeding retry limit
    at ChildProcessWorker.initialize (E:\arafat2.0\blog\node_modules\next\dist\compiled\jest-worker\index.js:1:11580)
    at ChildProcessWorker._onExit (E:\arafat2.0\blog\node_modules\next\dist\compiled\jest-worker\index.js:1:12545)
    at ChildProcess.emit (node:events:529:35)
    at ChildProcess._handle.onexit (node:internal/child_process:292:12) {
  type: 'WorkerError'
}

### Additional information

```js
PS E:\arafat2.0\blog> npm run build

> fati_blog@0.1.0 build
> next build

   ▲ Next.js 14.1.0
   - Environments: .env

   Creating an optimized production build ...
Compiler server unexpectedly exited with code: 3221225725 and signal: null

> Build error occurred
Error: Jest worker encountered 1 child process exceptions, exceeding retry limit
    at ChildProcessWorker.initialize (E:\arafat2.0\blog\node_modules\next\dist\compiled\jest-worker\index.js:1:11580)
    at ChildProcessWorker._onExit (E:\arafat2.0\blog\node_modules\next\dist\compiled\jest-worker\index.js:1:12545)
    at ChildProcess.emit (node:events:529:35)
    at ChildProcess._handle.onexit (node:internal/child_process:292:12) {
  type: 'WorkerError'
}
```

**Comment by hasan-mohammad-bd** [comment]
Here are some potential solutions:
1. Run npm run clean and then try building again with npm run build.
2. Ensure that your file system is not corrupted and that you have sufficient disk space.
3. Try setting swc to false in your next.config.js file.

**Reply by arafat813** [reply]
Thank you for your response. I tried as you suggested, but it still doesn't work, and I also have enough disk space.

**Reply by aaliosman** [reply]
i have baseUrl in .env file, change your localhost to 'your-IPV4:1337';     // Use IPv4 then restart your project   = solved

**Comment by SamOyeAH** [comment]
I am also encountering the same problem

**Reply by arafat813** [reply]
I have updated all packages using NPM update, and once this issue is resolved, you can also try it out

**Reply by SamOyeAH** [reply]
I have updated all my packages and I still receive the error unfortunately

**Reply by Ho3einWave** [reply]
did you manage to solve the issue?

**Comment by gabrielfernandes-codes** [comment]
That is the same you have @Gustav-J

**Comment by sban2009** [comment]
any updates on this?

**Reply by arafat813** [reply]
I have updated all packages using NPM update, and once this issue is resolved, you can also try it out

**Comment by abilitycoding** [comment]
i have same error please reply?  anybody find solution?

**Comment by Hyodori04** [comment]
me too same error

**Reply by ElisefRaz17** [reply]
Me as well. My team had to update our next.js application to this version and I still haven't found a solution to it.

**Comment by alirizasahinn** [comment]
I'm having the same issue... Couldn't find literally no help on the internet for this topic, whenever you search always jest the testing framework pops up.

**Comment by netpoe** [comment]
I had the same issue.
Vercel sets the NodeJS version to the latest automatically (currently at 22.+). Set the version to what you use in your local machine and it should run the deployment correctly. In my case I was developing in NodeJS 20+ locally but Vercel had 22+. 

If you set your NodeJS version in your machine to 22.+ and try to `next dev` your app, you should get the exact same errors as Vercel's.

**Reply by erichutabarat** [reply]
I had the same issue; I downgraded the Node version in the settings before redeploying, and it worked. more information [here](https://vercel.com/changelog/node-js-version-now-customizable-in-the-project-settings)

**Comment by to-abhinav** [comment]
This error might occur due to a misconfiguration of environment variables. Make sure to check your `.env` file. I faced a similar issue while deploying to Heroku because I had forgotten to add the required environment variables.


`
Static worker exited with code: 1 and signal: null 

 > Build error occurred
 Error: Jest worker encountered 4 child process exceptions, exceeding retry limit
 at ChildProcessWorker.initialize 
              (/tmp/build_cfaec33e/node_modules/next/dist/compiled/jest-worker/index.js:1:11580)
  at ChildProcessWorker._onExit 
              (/tmp/build_cfaec33e/node_modules/next/dist/compiled/jest-worker/index.js:1:12545)
  at ChildProcess.emit (node:events:524:28)
  at ChildProcess._handle.onexit (node:internal/child_process:293:12) 
{
  type: 'WorkerError'
}
`

**Comment by aaliosman** [comment]
i have baseUrl in .env file, change your localhost to 'your-IPV4:1337'; // Use IPv4 then restart your project = solved
